charts/generic-prometheus-alerts/ci/tests/ingress-alerts-tests.yaml

```diff
@@ -0,0 +1,170 @@
+        exp_alerts: []
+
+  ## alert:
+  ##   - RatelimitBlocking
+  ##
+  - interval: 1m
+    input_series:
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="429"}'
+        values: '0 1 3'
+
+    name: RatelimitBlocking alert fired as request rate increases over the duration of the 1 minute time range window
+
+    promql_expr_test:
+      # assert that
+      #   at 2m the rate (increase per second) in the last minute: 0.03 ((3 - 1) / 60)
+      #   which multiplied by 60 is greater than 0
+      #   the average is 2
+      - expr: avg(rate(nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status=~"429"}[1m]) * 60 > 0) by (ingress, exported_namespace)
+        eval_time: 2m
+        exp_samples:
+          - labels: '{exported_namespace="test-application-dev", ingress="test-application-ingress"}'
+            value: 2
+
+    alert_rule_test:
+      # assert that the RatelimitBlocking alert is firing
+      - eval_time: 2m
+        alertname: RatelimitBlocking
+        exp_alerts:
+          - exp_labels:
+              alertname: RatelimitBlocking
+              exported_namespace: test-application-dev
+              ingress: test-application-ingress
+              severity: alert-severity-tag
+            exp_annotations:
+              message: "Rate limit is being applied on ingress test-application-dev/test-application-ingress."
+              runbook_url: https://runbook.com/ingress-rate-limiting
+              dashboard_url: https://grafana.live.cloud-platform.service.justice.gov.uk/d/golden-signals/golden-signals?orgId=1&var-namespace=test-application-dev&var-service=test-application
+
+  - interval: 1m
+    input_series:
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="429"}'
+        values: '0 1 0'
+
+    name: RatelimitBlocking alert not fired as request rate decreases over the duration of the 1 minute time range window
+
+    promql_expr_test:
+      # at 2m the rate (increase per second) in the last minute: 0 and multiplying that by 60 is still 0 which is not greater than 0, assert nothing is returned
+      - expr: avg(rate(nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status=~"429"}[1m]) * 60 > 0) by (ingress, exported_namespace)
+        eval_time: 2m
+        exp_samples: []
+
+    alert_rule_test:
+      # assert that the RatelimitBlocking alert is NOT firing
+      - eval_time: 2m
+        alertname: RatelimitBlocking
+        exp_alerts: []
+
+  ## alert:
+  ##   - ModSecurityBlocking
+  ##
+  - interval: 1m
+    input_series:
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="406"}'
+        values: '0 21 42'
```

Comment:
Very minor, but if you made this `0 21 43` then it would be more obvious that the `value: 22` comes from the second minute and not the first.

Reply:
Great shout - addressed with https://github.com/ministryofjustice/hmpps-helm-charts/pull/79/commits/eb34cb103044f7e0175e7f06d957d42cba441272